autotest/gcore/vsiswift.py

```diff
@@ -297,6 +311,114 @@ def method(request):
         gdal.VSIFCloseL(f)
 
 
+###############################################################################
+# Test authentication with OS_IDENTITY_API_VERSION=3 OS_AUTH_TYPE="v3applicationcredential"
+# + OS_APPLICATION_CREDENTIAL_ID + OS_APPLICATION_CREDENTIAL_SECRET
+
+
+def test_vsiswift_fake_auth_v3_application_credential_url():
+
+    if gdaltest.webserver_port == 0:
+        pytest.skip()
+
+    gdal.VSICurlClearCache()
+    gdal.SetConfigOption('OS_IDENTITY_API_VERSION', '3')
```

Comment:
you should rather use 
```
with gdaltest.set_config_options( { 'OS_IDENTITY_API_VERSION': '3',
                                     'OS_AUTH_URL': '...', .... } ): 
```
so that options are set only in the given context and don't mess with later tests

Reply:
ah yes, right, of course, thanks for the tip.

Reply:
@rouault 
I'm now getting `AttributeError: module 'gdaltest' has no attribute 'set_config_options'`

Reply:
I worked it out. Correct code is `with gdal_test.config_options()`.